src/components/forms/FieldsCreator.tsx

```diff
@@ -0,0 +1,135 @@
+import sortBy from "lodash/sortBy"
+
+import { IFormVariable } from "../../utils/types"
+import { groupByOrderVariables } from "../../utils/forms"
+
+import StringField from "./fields/StringField"
+// import NumberField from "./fields/NumberField"
+import BooleanField from "./fields/BooleanField"
+import SelectField from "./fields/SelectField"
+// import ListField from "./fields/ListField"
+//import FilesField from "./fields/FilesField"
+import SelectRadioField from "./fields/SelectRadioField"
+import DateField from "./fields/DateField"
+
+import { useFormikContext, FormikValues } from "formik"
+import { useEffect, useState } from "react"
+
+interface FieldsCreatorProps {
+  variableList: IFormVariable[]
+  handleChange: Function
+  handleChangeInput: Function
+  handleValueChange: Function
+  handleTargetValueChange: Function
+}
+
+const FieldsCreator: React.FC<FieldsCreatorProps> = ({
+  variableList,
+  handleChange,
+  handleChangeInput,
+  handleValueChange,
+  handleTargetValueChange,
+}) => {
+  const sortedList = sortBy(variableList, "order")
+  const groupSortedList = groupByOrderVariables(sortedList)
+
+  const { values, setFieldValue } = useFormikContext<FormikValues>()
+  const [timer, setTimer] = useState<any>(null)
+
+  const onChangeHandle = async (e: React.ChangeEvent<HTMLInputElement>) => {
+    setFieldValue(e.target.name, e.target.value)
+
+    clearTimeout(timer)
+    if (e.target.value) {
+      const changeTimer = setTimeout(() => {
+        handleChangeInput(e)
+      }, 1000)
+      setTimer(changeTimer)
+    }
+  }
+
+  const handleChangeOption = (e: React.ChangeEvent<HTMLSelectElement>) => {
+    setFieldValue(e.target.name, e.target.value)
+    handleChange(e)
+  }
+
+  useEffect(() => {
+    handleValueChange(values)
+  }, [values])
+
+  const renderControls = (variable: IFormVariable) => {
+    switch (variable.type) {
+      case "string":
+        return (
+          <StringField variable={variable} onChangeHandle={onChangeHandle} />
+        )
+      // case "number":
+      //   return <NumberField variable={variable} />
+      case "bool":
+        return <BooleanField variable={variable} />
+      case "select":
+        return (
+          <SelectField variable={variable} handleChange={handleChangeOption} />
+        )
+      case "selectradio":
+        return (
+          <SelectRadioField
+            variable={variable}
+            handleChange={handleChangeOption}
+          />
+        )
+      case "dob":
+        return (
+          <DateField
+            variable={variable}
+            handleTargetValueChange={handleTargetValueChange}
+          />
+        )
+      // case "list(string)":
+      //   return <ListField variable={variable} />
+      // case "file":
+      //   return <FilesField variable={variable} />
+      default:
+        return (
+          <StringField variable={variable} onChangeHandle={onChangeHandle} />
+        )
```

Comment:
You have different prop names for the change handler. Please name all as `onChange`